fundrebalancer/fundrebalancer.go

```diff
@@ -674,9 +675,50 @@ func (r *FundRebalancer) isGasAcceptable(ctx context.Context, txns []SkipGoTxnWi
 		return false, "", fmt.Errorf("parsing max gas cost threshold")
 	}
 
-	if gasCostUUSDC.Cmp(maxCost) > 0 {
+	if gasCostUUSDC.Cmp(maxCost) <= 0 {
+		return true, gasCostUUSDC.String(), nil
+	}
+
+	// No fund rebalancing timeout set
+	if *chainConfig.FundRebalancingTimeout == -1 {
+		return false, gasCostUUSDC.String(), nil
+	}
+
+	transfers, err := r.database.GetPendingRebalanceTransfersToChain(ctx, chainID)
+	if err != nil {
+		return false, "", fmt.Errorf("getting pending transfers: %w", err)
+	}
+
+	if len(transfers) == 0 || chainConfig.FundRebalancingTimeout == nil {
 		return false, gasCostUUSDC.String(), nil
 	}
 
-	return true, gasCostUUSDC.String(), nil
+	// Use oldest pending transfer
+	oldestTransfer := transfers[0]
+	for _, transfer := range transfers {
+		if transfer.CreatedAt.Before(oldestTransfer.CreatedAt) {
+			oldestTransfer = transfer
+		}
+	}
+
+	// If timeout is exceeded, Use higher cost cap for timed out rebalancing
+	if time.Since(oldestTransfer.CreatedAt) > *chainConfig.FundRebalancingTimeout {
```

Comment:
I dont believe this has the desired effect. This logic uses the creation time of the oldest pending transfer, but a pending transfer has already landed on chain, so I'm not sure this gives the insight we want into whether the solver rebalance has been blocked on meeting the max cost check. We should do something like tracking in memory how long the rebalancer has been failing on profitability checks per chain and using that to determine the timeout.